sdk/web-pubsub/web-pubsub-express/src/cloudEventsDispatcher.ts

```diff
@@ -37,20 +43,28 @@ function getConnectResponseHandler(
       modified = true;
     },
     success(res?: ConnectResponse): void {
-      response.statusCode = 200;
       if (modified) {
         response.setHeader("ce-connectionState", utils.toBase64JsonString(states));
       }
       if (res === undefined) {
+        response.statusCode = 204;
         response.end();
       } else {
+        response.statusCode = 200;
         response.setHeader("Content-Type", "application/json; charset=utf-8");
         response.end(JSON.stringify(res));
       }
     },
     fail(code: 400 | 401 | 500, detail?: string): void {
-      response.statusCode = code;
-      response.end(detail ?? "");
+      handleConnectErrorResponse(connectRequest, response, code, detail);
+    },
+    failWith(res: ConnectErrorResponse | MqttConnectErrorResponse) {
+      if ("mqtt" in res) {
+        response.statusCode = getStatusCodeFromMqttConnectCode(res.mqtt.code);
```

Comment:
check if response.statuscode is set? If it is set, no need to read from mqtt.code

Reply:
Any particular reason why we need to check this? I assume if response status code is set by other middleware, it should have been returned right then before going to `web-pubsub-express` middleware